src/components/Select.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  useState,
+  useEffect,
+  useRef,
+  ReactNode,
+  SelectHTMLAttributes,
+} from "react";
+
+import { DropdownDown, DropdownUp } from "@/assets/icon";
+import { cn } from "@/utils/cn";
+
+const sizeMap = {
+  lg: "py-4 px-5 text-[1rem]",
+  sm: "p-2.5 text-sm",
+} as const;
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface SelectProps
+  extends Omit<
+    SelectHTMLAttributes<HTMLButtonElement>,
+    "size" | "onChange" | "disabled"
+  > {
+  id?: string;
+  label?: string;
+  options: Option[];
+  value?: string;
+  onValueChange?: (value: string) => void;
+  placeholder?: string;
+  size?: keyof typeof sizeMap;
+  fullWidth?: boolean;
+  width?: number | string;
+  className?: string;
+  wrapperClassName?: string;
+}
+
+// 라벨과 입력 영역을 감싸는 공통 컴포넌트
+function Field({
+  id,
+  label,
+  children,
+}: {
+  id?: string;
+  label?: string;
+  children: ReactNode;
+}) {
+  return (
+    <div>
+      {label && (
+        <label htmlFor={id} className="inline-block mb-2 leading-[1.625rem]">
+          {label}
+        </label>
+      )}
+      {children}
+    </div>
+  );
+}
+
+function Select({
+  id,
+  label,
+  options,
+  value,
+  onValueChange,
+  placeholder = "선택",
+  size = "lg",
+  fullWidth,
+  width,
+  className,
+  wrapperClassName,
+  ...rest
+}: SelectProps) {
+  const [open, setOpen] = useState(false);
+  const [buttonWidth, setButtonWidth] = useState<number>(0);
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const selectedOption = options.find((option) => option.value === value);
+
+  const wrapperClassNames = cn(
+    "relative",
+    {
+      "w-full": fullWidth,
+    },
+    wrapperClassName,
+  );
+
+  const buttonClassNames = cn(
+    "flex items-center justify-between rounded-[0.375rem]",
+    {
+      "w-full": fullWidth,
+      "bg-white placeholder:text-gray-40 border border-gray-30": size === "lg",
+      "bg-gray-10 font-bold": size === "sm",
+    },
+    sizeMap[size],
+    className,
+  );
+
+  const listClassNames = cn(
+    "absolute top-full left-0 mt-1 border rounded-[0.375rem] bg-white border-gray-30 shadow-lg z-10 max-h-48 overflow-y-auto",
+  );
```

Comment:
```suggestion
  const listClassNames = cn(
    "absolute top-full left-0 mt-1 border rounded-[0.375rem] bg-white border-gray-30 text-black shadow-lg z-10 max-h-48 overflow-y-auto",
  );
```

리스트 아이템에 `text-black` 스타일이 적용되어 있지 않아요.
현재는 `#000000`이지만, 수민님께서 등록해주신 테마 색상인 `text-black`은 엄연히 `#111322`로 다르니까요! 👍

텍스트와 관련된 속성은 대부분 하위 자식요소까지 상속되기 때문에 리스트에 클래스명을 주면 좋을 것 같습니다! 😄